### PR TITLE
fix: Integration of Navigation Architectural Components

### LINKED ISSUE #966: Integration of Navigation Architecture Component
### Feature Request

**Feature Description**
This is aimed at integrating Navigation Component in the app in order to simplify the navigation process and make it less buggy.
https://developer.android.com/topic/libraries/architecture/navigation/
   
**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Masquerade0097**
Fixes #966 

Changes: 

Not ready for review.

Screenshots for the change:

### CODE REVIEW

**Comment by Masquerade0097** [comment]
@iamareebjamal 
I'm testing Nav compoenents for AuthActivity first.
I'm populating the NavHostFragment in AuthActivity with Nav Graph on start and the app crashes with this error.  
This is the stacktrace.
`E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.fossasia.eventyay, PID: 15597
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.fossasia.eventyay/org.fossasia.openevent.app.core.auth.AuthActivity}: java.lang.IllegalStateException: Activity org.fossasia.openevent.app.core.auth.AuthActivity@247f5 does not have a NavController set on 2131296433
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2659)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2724)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1473)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6123)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
     Caused by: java.lang.IllegalStateException: Activity org.fossasia.openevent.app.core.auth.AuthActivity@247f5 does not have a NavController set on 2131296433
        at androidx.navigation.Navigation.findNavController(Navigation.java:60)
        at org.fossasia.openevent.app.core.auth.AuthActivity.onCreate(AuthActivity.java:44)
        at android.app.Activity.performCreate(Activity.java:6672)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1140)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2612)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2724) 
        at android.app.ActivityThread.-wrap12(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1473) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6123) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757) `

When I open AuthActivity using old Fragment transaction logic then the loginFragment opens. Then I'm using Navvigation components for signup button to transition to signupfragement but again I'm gettign this error that `NavController is not set`.

File: app/src/main/res/layout/main_app_bar.xml [hunk 1]
```diff
@@ -25,7 +25,10 @@
             android:id="@+id/fragment_container"
             android:layout_width="match_parent"
             android:layout_height="match_parent"
-            app:layout_behavior="@string/appbar_scrolling_view_behavior" />
+            app:layout_behavior="@string/appbar_scrolling_view_behavior"
+            android:name="android.navigation.fragment.NavHostFragment"
+            app:navGraph="@navigation/main_navigation"
+            app:defaultNavHost="true"/>
```
Comment (iamareebjamal) [inline app/src/main/res/layout/main_app_bar.xml]: This should be in AuthActivity as nav graph of Auth will be different from main nav

File: app/src/main/res/navigation/main_navigation.xml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (iamareebjamal) [inline app/src/main/res/navigation/main_navigation.xml]: Should be auth_navigation

File: app/src/main/res/layout/main_app_bar.xml (hunk 1 above)
Comment (Masquerade0097) [reply app/src/main/res/layout/main_app_bar.xml]: @iamareebjamal I already tried incorporating the app_bar code in auth_activity but still the error persists.

**Comment by Masquerade0097** [comment]
The Navigation Component have issues with fragment transitions animations along with others as discussed in #1011 .Therefore, closing this for now.

**Pull request closed without merging**